src/utils.ts

```diff
@@ -0,0 +1,759 @@
+// Dependencies
+import chalk from 'chalk';
+import fs from 'fs';
+import path from 'path';
+import Fuse from 'fuse.js';
+import ErrorWithoutStack from './error-without-stack';
+
+// Interface declarations
+interface InputObject {
+	command: string;
+	data?: string | number;
+	[propName: string]: boolean | string | number | undefined;
+}
+
+// Interface declarations
+export interface AppSettings {
+	name: string | null;
+	packageName: string | null;
+	version: string | null;
+	[propName: string]: string | number | null;
+}
+
+interface CommandSpecOption {
+	accepts?: string[];
+	cascades?: boolean;
+	description?: string;
+	required?: boolean;
+	shorthand?: string;
+	type?: string;
+}
+
+interface CommandSpecOptions {
+	[propName: string]: CommandSpecOption;
+}
+
+interface CommandSpecFlag {
+	cascades?: boolean;
+	description?: string | null;
+	shorthand?: string;
+}
+
+interface CommandSpecFlags {
+	[propName: string]: CommandSpecFlag;
+}
+
+interface CommandSpecData {
+	accepts?: string[];
+	description?: string;
+	ignoreFlagsAndOptions?: boolean;
+	required?: boolean;
+	type?: string;
+}
+
+export interface CommandSpec {
+	data?: CommandSpecData;
+	description?: string | null;
+	flags: CommandSpecFlags;
+	options: CommandSpecOptions;
+	executeOnCascade?: boolean;
+}
+
+interface ConstructorSettingsSpacing {
+	after?: number;
+	before?: number;
+}
+
+interface NewVersionWarningSettings {
+	enabled: boolean;
+	installedGlobally: boolean;
+}
+
+export interface ConstructorSettings {
+	app?: AppSettings;
+	arguments?: string[];
+	mainFilename: string;
+	newVersionWarning?: NewVersionWarningSettings;
+	onStart?: string | Function | null;
+	packageFilePath?: string;
+	spacing?: ConstructorSettingsSpacing;
```

Comment:
I would eliminate a separate `ConstructorSettingsSpacing` interface and simply define this directly here:

```suggestion
	spacing?: {
		before?: number;
		after?: number;
	};
```